source/mir/random/tinflex/internal/area.d

```diff
@@ -0,0 +1,650 @@
+    // sigma in the paper (1: left side, -1, right side
+    const byte leftOrRight = (iv.rx - sh._y) > (sh._y - iv.lx) ? 1 : -1;
+
+    // sh.y is the boundary point where f obtains its maximum
+
+    // specializations for T_c family (page 6)
+    if (iv.c == 0)
+    {
+        // T_c = log(x)
+        // Error in table, see equation (4)
+        immutable z = leftOrRight * sh.slope * (iv.rx - iv.lx);
+        // check whether approximation is possible, page 5
+        if (abs(z) < S(1e-6))
+        {
+            area = exp(sh.a) * (iv.rx - iv.lx) * (1 + z / 2 + (z * z) / 6);
+        }
+        else
+        {
+            // F_T = e^x
+            area = (exp(sh(iv.rx)) - exp(sh(iv.lx))) / sh.slope;
+        }
+    }
+    else
+    {
+        // for c < 0, the tangent result must result in a valid (bounded) hat function
+        if (iv.c * sh(iv.rx) < 0 || iv.c * sh(iv.lx) < 0)
+        {
+            // returning infinity will yield a split on this interval.
+            area = S.max;
+            goto L;
+        }
+
+        immutable intLength = iv.rx - iv.lx;
+        immutable z = leftOrRight * sh.slope / sh.a * intLength;
+
+        if (iv.c == 1)
+        {
+            // T_c^-1 = x^c
+            area = S(0.5) * sh.a * intLength * (z + 2);
+        }
+        else if (iv.c == S(-0.5))
+        {
+            // T_c = -1/sqrt(x)
+            if (abs(z) < S(0.5))
+            {
+                // T_c^-1 = 1/x^2
+                area = 1 / (sh.a * sh.a) * intLength / (1 + z);
+            }
+            else
+            {
+                area = ((-1 / sh(iv.rx)) + (1 / sh(iv.lx))) / sh.slope;
+            }
+        }
+        else if (iv.c == -1)
+        {
+            // T_C = -1 / x
+            if (abs(z) <= S(1e-6))
+            {
+                // T_C^-1 = -1 / x
+                area = -1 / sh.a * intLength * (1 - z / 2 + z * z / 3);
+            }
+            else
+            {
+                // F_T = -log(-x)
+                area = (-log(-sh(iv.rx)) + log(-sh(iv.lx))) / sh.slope;
+            }
+        }
+        else
+        {
+            // T_c = -1 / x
+            //area = (r - l) * c / (c + 1) * 1 / z * ((1 + z)^^((c + 1) / c) - 1);
+            if (abs(sh.slope) > S(1e-10))
+            {
+                alias ad = antiderivative;
+                area = (ad(sh(iv.rx), iv.c) - ad(sh(iv.lx), iv.c)) / sh.slope;
+            }
+            else
+            {
+                area = inverse(sh.a, iv.c) * intLength;
+            }
+        }
+    }
+
+L:
+    // if we receive an invalid value, we require the interval to be split
+    import std.math : isFinite;
+    if (!isFinite(area))
+        area = S.max;
+    else if (area < 0)
+        area = S.max;
+
+    static if (!isHat)
+    {
+        // squeeze may return infinity
+        if (area == S.max)
+        {
+            area = 0;
+        }
+    }
+
+    static if (isHat)
+        iv.hatArea = area;
+    else
+        iv.squeezeArea = area;
```

Comment:
These `static if` statements related to `isHat` can probably be combined into one `static if/else` block.